synapse/logging/opentracing.py

```diff
@@ -823,75 +824,110 @@ def extract_text_map(carrier: Dict[str, str]) -> Optional["opentracing.SpanConte
 # Tracing decorators
 
 
-def trace_with_opname(opname: str) -> Callable[[Callable[P, R]], Callable[P, R]]:
+def _create_decorator(
+    func: Callable[P, R],
+    # TODO: What is the correct type for these `Any`? `P.args, P.kwargs` isn't allowed here
+    wrapping_logic: Callable[[Callable[P, R], Any, Any], ContextManager[None]],
```

Comment:
I need some assistance with the types here and a few other places 🙏, https://github.com/matrix-org/synapse/runs/7664256706?check_suite_focus=true

Reply:
Thanks for the help so far @squahtx! Still a few more problem spots: https://github.com/matrix-org/synapse/runs/7681624995?check_suite_focus=true

```
synapse/logging/opentracing.py:914: error: Argument 1 to "contextmanager" has incompatible type "Callable[[Callable[P, R], **P], Generator[None, None, None]]"; expected "Callable[[Callable[P, R], **P], Iterator[None]]"  [arg-type]
synapse/logging/opentracing.py:953: error: Argument 1 to "contextmanager" has incompatible type "Callable[[Callable[P, R], **P], Generator[None, None, None]]"; expected "Callable[[Callable[P, R], **P], Iterator[None]]"  [arg-type]
```

Reply:
Those last two errors look like a bug in mypy:
Changing `Generator[...]` to `Iterator[None]` doesn't fix them.
Changing `P.args` and `P.kwargs` to `Any` makes them go away.

I'm in favor of slapping a `# type: ignore[arg-type]`s on, with a link to the mypy bug (likely https://github.com/python/mypy/issues/12909)

Reply:
What's a good way to add a link to the bug? The following just says `Invalid "type: ignore" comment`

```
# type: ignore[arg-type] https://github.com/python/mypy/issues/12909
```

So far I've just put a `# mypy bug: https://github.com/python/mypy/issues/12909` above

Reply:
We've not got any particular convention. As long as the url is in a comment nearby it's good enough!